File: Java/src/com/dbc/algorithm/Daily/SwappingNodesInALinkedList.java
package com.dbc.algorithm.Daily;

public class SwappingNodesInALinkedList {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode swapNodes(ListNode head, int k) {
        int count = 0;
        ListNode node = head;
        ListNode point = head;

        while (point != null) {
            count++;
            if (count == k) {
                node = point;
            }
            point = point.next;
        }

        point = head;
        for (int i = 0; i < count - k; i++) {
            point = point.next;
        }

        int temp = node.val;
        node.val = point.val;
        point.val = temp;

        return head;
    }
}

File: core/tmp/Python/linked-list/swapping_nodes_in_a_linked_list.py
class ListNode:
    def __init__(self, val=0, next=None) -> None:
        super().__init__()
        self.val = val
        self.next = next


def swapNodes(head: ListNode, k: int) -> ListNode:
    count = 0
    point = head
    node = None

    while point is not None:
        count += 1
        if count == k:
            node = point
        point = point.next
    point = head
    for i in range(count - k):
        point = point.next

    temp = point.val
    point.val = node.val
    node.val = temp

    return head
